README.md

```diff
@@ -1,37 +1,79 @@
 
-This project contains windows programs to isolate a program in an own desktop to prevent malware (without OS privileges) to capture user input or sreenshots.
-It consists of 4 sub-programs: The `CageService`, `CageManager`, `StarterCmd`, and `ImageSelectDialog`. The 3 actual parts (Service, Manager and StarterCmd) communicate over a TCP connection to interchange messages.
+This project contains Windows programs to isolate a user specific program in an own desktop to prevent malware (without OS privileges) to capture user input or screenshots.
+After the installation of Shark Cage, create a config using the `CageConfigurator` containing the program (e.g. Firefox for online banking) which should run in a secure environment, optionally an additional programm (e.g. Keepass to retrieve the password for the online banking) and an icon which is later used to identify, that the program was started in the secure environment.
+
+All programs running in the Shark Cage will be started on a further, isolated desktop which a malware without administrator privilieges can not access.
 
 ## Components
 
+This project consists of five sub-programs:
+
+* `CageService`
+* `CageManager`
+* `CageChooser`
+* `CageConfigurator`
+* `SharkCageInstaller`
+
+ The three parts (Service, Manager and Configurator) communicate over a TCP connection to interchange messages.
+
 ### CageService
 
-The `CageService` implements a Windows service running the background. It receives messages from the UI and sends messages to the `CageManager`.
+The `CageService` implements a Windows service running in the background. It receives messages from the `CageChooser` and sends messages to the `CageManager`.
 
 ### CageManager
 
-The `CageManager` creates a new desktop and processes in it, according to the messages received by the `CageService`.
+The `CageManager` creates a new desktop and starts the program and optionally the additional program according to the config received from the `CageChooser` over the `CageService`. In addition, the icon and some additional informations are displayed. Using the displayed "Activate"-button(s) the program(s) can be restarted or brought back into the foreground.
+
+<details><summary markdown="span"><code>Fold out to see sample</code></summary>
+
+<img width="1680" alt="Screenshot CageManager" src="https://user-images.githubusercontent.com/1786772/43678602-914198ec-9816-11e8-8a75-0ac3032368e4.png">
+
+</details>
+
+### CageChooser
+
+The `CageChooser` is a user interface which displays in a list all available configs on the system by iterating over the registry entries under the following path: `Computer\HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\SharkCage\Configs`. By selecting a config and pressing the "Start"-button (or the enter key), the `CageManager` will be started. As soon as the `CageManager` is started, the `CageChooser` will send the path to the selected config to the `CageManager`.
```

Comment:
* '[...] entries ~under~ at the following path: [...]'
* last part: The creation of the CageManager is done implicitely when receiving a `START_PROCESS` message which means the Chooser will only send one message with the config path and then everything else happens automatically